### PR TITLE
Fixes #426 Refactored InvocationMarker to a static utility class

### LINKED ISSUE #426: Refactor instance based utility classes to static utility classes
The internal implementation of Mockito contains numerous helper classed that serve as a container for useful methods. By its nature these classes don't have a state. Currently some them are instance based in other word you need to create the utility class to call a helper method. This not only pollutes the heap but also the code cause instance methods can't be imported statically. 

Here is an example:

```java
Helper helper = new Helper();
if (helper.isInputValid(input)){
  [...] 
}
```

vs. static import of Helper.isInputValid

```java
if (isInputValid(input)){
  [...] 
}
```

The aim of this ticket is to identify canidates that can be refactored to static utility classes. If you like to refactoring and mockito feel free to send a PR and reference this issue.

Refactoring canidates:
- [ ] `AccessibilityChanger`
- [ ] `BeanPropertySetter`
- [ ] `ConditionalStackTraceFilter`
- [ ] `FieldCopier`
- [ ] `FieldReader`
- [ ] `GenericMaster` should be integrate into `GenericTypeResolver`
- [ ] `JUnitFailureHacker` can be removed when the deprecated `VerboseMockitoJUnitRunner` is removed
- [ ] `LenientCopyTool`
- [ ] `MatcherBinder`
- [ ] `MockitoCore` should better be a singleton
- [ ] `MockCreationValidator`
- [ ] `RemoveFirstLine`
- [x] #591 `ArgumentMatchingTool`
- [x] #515 `AllInvocationsFinder`
- [x] #502 `ArgumentsComparator`
- [x] #540 `ArrayUtils`
- [x] #490 `AtLeastXNumberOfInvocationsChecker`
- [x] #490 `AtLeastXNumberOfInvocationsInOrderChecker`
- [ ] #912 `Constructors`
- [x] #427 `FieldSetter`
- [x] #908 `FriendlyExceptionMaker`
- [x] #431 `HandyReturnValues` 
- [x] #432 `InvocationMarker`
- [x] #462 `InvocationsFinder`
- [x] #908 `JUnitDetecter`
- [x] #490 `MissingInvocationChecker`
- [x] #490 `MissingInvocationInOrderChecker`
- [x] #514 `MockUtil`
- [x] #503 `NonGreedyNumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsChecker`
- [x] #547 `ObjectMethodsGuru`
- [x] #427 `Reporter`
- [x] #535 `SuperTypesLastSorter`
- [x] #501 `TestMethodFinder`
- [x] #515 `VerifiableInvocationsFinder`

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
#426 

Refactored `InvocationMarker `to a static utility class

### CODE REVIEW

File: src/main/java/org/mockito/internal/invocation/InvocationMarker.java [hunk 1]
```diff
@@ -11,18 +11,20 @@
 
 public class InvocationMarker {
 
-    public void markVerified(List<Invocation> invocations, CapturesArgumentsFromInvocation wanted) {
+	private InvocationMarker(){}
```
Comment (szpak) [inline src/main/java/org/mockito/internal/invocation/InvocationMarker.java]: Formatting again?

File: dummy-commit.txt [hunk 2]
```diff
@@ -1 +1 @@
-Change this if you need a dummy commit. Push build.
\ No newline at end of file
+Change this if you need a dummy commit. Push build.
```
Comment (szpak) [inline dummy-commit.txt]: ?

Comment (ChristianSchwarz) [inline dummy-commit.txt]: I added a new line in order to trigger a new build. The concurrent test are failing on every 2nd run.

File: src/main/java/org/mockito/internal/invocation/InvocationMarker.java (hunk 1 above)
Comment (ChristianSchwarz) [inline src/main/java/org/mockito/internal/invocation/InvocationMarker.java]: tab vs. 4x space , fixed...

**Comment by TimvdLippe** [comment]
There are some imports remaining on the `InvocationMarker` which I think can be removed. Other than that LGTM.

**Comment by ChristianSchwarz** [comment]
How to trigger a build without the commiting, squashing, forced push cycle? That failing tests suck!

**Comment by PascalSchumacher** [comment]
Not much faster but the fastest way I know of to trigger building of pull request is

```
git commit --amend
git push -f
```

**Comment by TimvdLippe** [comment]
LGTM

**Pull request merged**